src/psyclone/psyir/backend/fortran.py

```diff
@@ -1214,6 +1215,12 @@ def binaryoperation_node(self, node):
             fort_oper = self.get_operator(node.operator)
             if self.is_intrinsic(fort_oper):
                 # This is a binary intrinsic function.
+                # pylint: disable=unidiomatic-typecheck
+                if fort_oper in ["LBOUND", "UBOUND"] and \
```

Comment:
I don't like the look of this. Why is it necessary (given that the transformation now forbids modifying the argument to such operators)? Also, what happens if the argument to the operator is a reference to an array member of a structure?

Reply:
Removed changes to backend.